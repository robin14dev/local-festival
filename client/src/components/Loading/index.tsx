import React from "react";
import styled from "styled-components";
import loadMp4 from "../../assets/loading.mp4";

export const Wrapper = styled.article`
  margin-left: auto;
  margin-right: auto;
  display: flex;
  flex-flow: column;
  justify-content: center;
  align-items: center;
  video {
    width: 20%;
  }

  div {
    color: gray;
    font-size: 1.2rem;
  }
`;

type LoadingProps = {
  text: string;
};

const Loading = ({ text }: LoadingProps) => {
  return (
    <Wrapper>
      <video
        data-testid="loading-video"
        autoPlay
        loop
        muted
        playsInline
        width="100%"
        height="100%"
      >
        <source src={loadMp4} type="video/mp4" />
      </video>
      <div>{text}</div>.
    </Wrapper>
  );
};

export default Loading;
