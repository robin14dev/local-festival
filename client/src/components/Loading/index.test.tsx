import { render } from "@testing-library/react";
import React from "react";
import Loading from ".";
import "jest-styled-components";

describe("<Loading />", () => {
  it("텍스트를 포함하여 컴포넌트를 렌더링합니다", () => {
    const text = "Loading...";
    const { getByText, container } = render(<Loading text={text} />);

    expect(getByText(text)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("비디오 요소를 로딩 컴포넌트에 포함시켜 렌더링합니다.", () => {
    const text = "Loading...";
    const { getByTestId } = render(<Loading text={text} />);

    const videoElement = getByTestId("loading-video");
    expect(videoElement).toBeInTheDocument();
    expect(videoElement).toHaveAttribute("autoPlay");
    expect(videoElement).toHaveAttribute("loop");
    expect(videoElement).toHaveAttribute("playsInline");
    expect(videoElement).toHaveAttribute("width", "100%");
    expect(videoElement).toHaveAttribute("height", "100%");
  });
});
