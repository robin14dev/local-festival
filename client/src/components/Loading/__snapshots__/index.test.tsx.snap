// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> renders with the provided text 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 video {
  width: 20%;
}

.c0 div {
  color: gray;
  font-size: 1.2rem;
}

<div>
  <article
    class="c0"
  >
    <video
      autoplay=""
      data-testid="loading-video"
      height="100%"
      loop=""
      playsinline=""
      width="100%"
    >
      <source
        src="loading.mp4"
        type="video/mp4"
      />
    </video>
    <div>
      Loading...
    </div>
    .
  </article>
</div>
`;

exports[`<Loading /> 텍스트를 포함하여 컴포넌트를 렌더링합니다 1`] = `
.c0 {
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 video {
  width: 20%;
}

.c0 div {
  color: gray;
  font-size: 1.2rem;
}

<div>
  <article
    class="c0"
  >
    <video
      autoplay=""
      data-testid="loading-video"
      height="100%"
      loop=""
      playsinline=""
      width="100%"
    >
      <source
        src="loading.mp4"
        type="video/mp4"
      />
    </video>
    <div>
      Loading...
    </div>
    .
  </article>
</div>
`;
